src/ucp/wireup/address.c

```diff
@@ -506,30 +507,36 @@ ucp_address_unpack_byte_extended(const void *ptr, size_t value_mask,
     return (void*)ptr;
 }
 
+static size_t ucp_address_md_mask(ucp_object_version_t addr_version)
+{
+    return (addr_version == UCP_OBJECT_VERSION_V1) ?
+           UCP_ADDRESS_FLAG_MD_MASK_V1 : UCP_ADDRESS_FLAG_MD_MASK;
+}
+
 static void *
 ucp_address_pack_md_info(void *ptr, int is_empty_dev, uint64_t md_flags,
                          ucp_md_index_t md_index,
                          ucp_object_version_t addr_version)
 {
-    void *flags_ptr = ptr;
-
-    ptr = ucp_address_pack_byte_extended(ptr, md_index,
-                                         UCP_ADDRESS_FLAG_MD_MASK,
-                                         addr_version);
+    uint8_t *flags_ptr = ptr;
```

Comment:
shall we do:
```
uint8_t *flags_ptr= ucs_serialize_next(ptr, uint8_t);
```
otherwise it would be overwritten in `ucp_address_pack_byte_extended()`